libraries/AP_Vehicle/AP_Vehicle.cpp

```diff
@@ -356,6 +356,15 @@ void AP_Vehicle::setup()
 #endif
 
 #if AP_SERIALMANAGER_ENABLED
+#if HAL_WITH_IO_MCU && (HAL_UART_IOMCU_IDX < HAL_UART_NUM_SERIAL_PORTS)
+    if (BoardConfig.io_enabled()) {
+        // If IO is enabled, we need to set the protocol to None
+        // as the IO MCU UART is handled by the AP_IOMCU class
+        serial_manager.set_protocol_and_baud(HAL_UART_IOMCU_IDX, AP_SerialManager::SerialProtocol_None, 0);
+    } else {
+        iomcu.wipe();
```

Comment:
why are we wiping the fw if IO not enabled?

Reply:
This is to allow flashing a different firmware to secondary. We have to reduce the boot time if IOMCU is enabled to 200ms, which is not long enough for it to be caught over USB. To allow updating Secondary MCU over USB again we need wipe iofirmware.

Reply:
Not relevant anymore, moved to different procedure to allow updates other than iofirmware.